examples/fool-patchguard.c

```diff
@@ -28,22 +28,115 @@
 #include <inttypes.h>
 #include <signal.h>
 
+#include <bddisasm/disasmtypes.h>
+#include <bddisasm/bddisasm.h>
+
 #include <libvmi/libvmi.h>
 #include <libvmi/events.h>
 
+// maximum size of an x86 instruction
+#define MAX_SIZE_X86_INSN 15
+#define KISERVICE_ENTRY_SIZE sizeof(uint32_t)
+
+// we use emul_read with 'dont_free' so it can't be allocated on the stack
+// as it could be corrupted
+static emul_read_t g_emul_read;
```

Comment:
` = { .dont_free = 1 };`

Reply:
I'm also thinking that it's perhaps time to change this ABI a bit and instead of having a `dont_free` field, just supply a pointer to a `free` function if one needs to be called.